scripts/batch-score.coffee

```diff
@@ -0,0 +1,50 @@
+# Description:
+#   Script for batch wise score.
+#
+# Commands:
+#   hubot score bxx
+# 
+# Author:
+#   aseem09
+
+cron = require('node-cron')
+module.exports = (robot) ->
+
+  listOfUsersWithCount = (year) ->
+  
+    sorted = []
+    for own key, user of robot.brain.data.users
+
+      temp2=user[4]
+      user_year=temp2.split('')
+      temp3 = parseInt( user_year[0], 10 );
+
+      if `temp3 == year` > 0
+        sorted.push [user.name, user.msgcount]
+
+    if sorted.length
+      sorted.sort (a, b) ->
+        b[1] - a[1]
+    return sorted
+  
+  # response for score status of any <batch>
+  robot.respond /score b([\w\-_]+)/i, (msg) ->
+
+    response = "```Name : Message Count\n"
+
+    #passing year of current first
+    relative_year=23
+
+    #changing the relative at the end of the year
+    cron.schedule '0 0 9 7 *', ()->
+      `relative_year = relative_year + 1`
```

Comment:
This method to get the year is very fragile! It will cause mapping errors after we restart the script! Also fixing a particular date will create confusion among the users

Reply:
I have done some changes. Please have a look